src/messages/FileMessage.vue

```diff
@@ -1,21 +1,60 @@
 <template>
-  <div class="sc-message--file" :style="messageColors">
-    <div class="sc-message--file-icon">
-      <img :src="data.file.url" class="sc-image" />
+  <div class="sc-message--file" :style="messageColors" :id="data.file.id">
+    <div class="sc-message--file-icon" v-if="data.file.mime == 'application/vnd.openxmlformats-officedocument.wordprocessingml.document'">
+      <ApplicationWord />
     </div>
-    <div class="sc-message--file-name" :style="messageColors">
-      <a :href="data.file.url ? data.file.url : '#'" target="_blank">{{ data.file.name || '' }}</a>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'">
+      <ApplicationExcel />
     </div>
-    <div class="sc-message--file-text" :style="messageColors">
-      {{ data.text }}
-      <p v-if="data.meta" class="sc-message--meta" :style="messageColors">
-        {{ data.meta }}
-      </p>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'application/vnd.openxmlformats-officedocument.presentationml.presentation'">
+      <ApplicationPowerPoint />
+    </div>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'application/zip'">
+      <ApplicationZip />
+    </div>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'image/jpeg'">
+      <ApplicationJpg />
+    </div>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'image/svg+xml'">
+      <ApplicationSvg />
+    </div>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'text/csv'">
+      <ApplicationCsv />
+    </div>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'application/pdf'">
+      <ApplicationPdf />
+    </div>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'image/png'">
+      <ApplicationPng />
+    </div>
+    <div class="sc-message--file-icon" v-else="">
+      <ApplicationDefault />
+    </div>
+    <div class="sc-message--file-content">
+      <div class="sc-message--file-name" :style="messageColors">
+        {{truncateFilename(data.file.name, 20) || ''}}
+      </div>
+      <DownloadIcon />
+      <div class="sc-message--file-size" :style="messageColors">
+        {{humanReadableFileSize(data.file.size) || ''}}
```

Comment:
I think this fails if data has no file property, `{{data.file && humanReadableFileSize(data.file.size || 0)}}` should be better but also ugly (user would see '0B' which is not the best).